drivers/ieee802154/Kconfig.esp32

```diff
@@ -0,0 +1,205 @@
+# Espressif ESP32 802.15.4 configuration options
+
+# Copyright (c) 2024 A Labs GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig IEEE802154_ESP32
+	bool "ESP32 series IEEE 802.15.4 Driver"
+	default y
+	depends on DT_HAS_ESPRESSIF_ESP32_IEEE802154_ENABLED
+
+if IEEE802154_ESP32
+
+config IEEE802154_ESP32_INIT_PRIO
+	int "ESP32 IEEE 802.15.4 initialization priority"
+	default 80
+	help
+	  Set the initialization priority number. Do not mess with it unless
+	  you know what you are doing.
+
+# Kconfigs copied from Espressif HAL module (ESP-IDF) below
+
+config IEEE802154_RX_BUFFER_SIZE
+	int "Number of 802.15.4 receive buffers"
+	default 20
+	range 2 100
+	help
+	  The number of 802.15.4 receive buffers.
+
+	  This config is used in the Espressif HAL module.
+
+choice IEEE802154_CCA_MODE
+	prompt "Clear Channel Assessment (CCA) mode"
+	default IEEE802154_CCA_ED
+	help
+	  Configure the CCA mode
+
+	  This config is used in the Espressif HAL module.
+
+config IEEE802154_CCA_CARRIER
+	bool "Carrier sense only"
+	help
+	  Configure the CCA mode to Carrier sense only
+
+config IEEE802154_CCA_ED
+	bool "Energy above threshold"
+	help
+	  Configure the CCA mode to Energy above threshold
+
+config IEEE802154_CCA_CARRIER_OR_ED
+	bool "Carrier sense OR energy above threshold"
+	help
+	  Configure the CCA mode to Carrier sense OR energy above threshold
+
+config IEEE802154_CCA_CARRIER_AND_ED
+	bool "Carrier sense AND energy above threshold"
+	help
+	  Configure the CCA mode to Carrier sense AND energy above threshold
+
+endchoice # IEEE802154_CCA_MODE
+
+config IEEE802154_CCA_MODE
+	int
+	default 0 if IEEE802154_CCA_CARRIER
+	default 1 if IEEE802154_CCA_ED
+	default 2 if IEEE802154_CCA_CARRIER_OR_ED
+	default 3 if IEEE802154_CCA_CARRIER_AND_ED
+
+config IEEE802154_CCA_THRESHOLD
+	int "CCA detection threshold"
+	range -120 0
+	default -60
+	help
+	  Set the CCA threshold, in dB.
+
+	  This config is used in the Espressif HAL module.
+
+config IEEE802154_PENDING_TABLE_SIZE
+	int "Pending table size"
+	range 1 100
+	default 20
+	help
+	  set the pending table size
+
+config IEEE802154_MULTI_PAN_ENABLE
+	bool "Multi-pan feature for frame filter"
+	help
+	  Enable IEEE802154 multi-pan
+
+	  This config is used in the Espressif HAL module.
+
+config IEEE802154_TIMING_OPTIMIZATION
```

Comment:
This entry only makes sense if a couple of functions are placed in IRAM. If it is not, I would remove it.